src/index.js

```diff
@@ -33,6 +33,8 @@ import ZipkinB3TextMapCodec from './propagators/zipkin_b3_text_map_codec';
 import TestUtils from './test_util';
 import TChannelBridge from './tchannel_bridge';
 
+import PrometheusMetricsFactory from './metrics/prometheus';
```

Comment:
isn't this going to blow up in applications that do not have prom_client dependency?

Reply:
bump

Reply:
I fixed this by not exporting PrometheusMetricsFactory if prom-client is not installed.
```node
try {
  require.resolve('prom-client');
  modules.PrometheusMetricsFactory = require('./metrics/prometheus').default;
} catch (e) {}
```
But IMO the code looks bad in terms of readability because require.resolve() just throws an exception if prom-client is not installed. Would it be better if making a function like [this](https://github.com/eundoosong/jaeger-client-node/blob/f868f53ceb0736a7ba3be7247e990ad0a13195b5/src/util.js#L164), or any better idea?

> I think there is a way to use the latest prom client and still maintain 0.10 compat. Can you do conditional import of the prom client, e.g. as shown here? In that case, the jaeger library itself does not need to have an explicit dependency on the prom client. When testing under 6+ version, we can npm install prom-client manually in the travis build and test the metrics factory. When running with 0.10 the test file can skip the tests. I've used this approach recently in the python client to support some tests only working in py-2.6, and not in py-3

As you told, I tried to use prometheus metrics with the latest prom-client only for node 6+, not for 0.10.  But it's a little bit tricky. 
In travis build(.travis.yml), see PROM_METRICS_TEST is added. With this flag, prom-client installation and the test are done if node is 6+.
https://github.com/eundoosong/jaeger-client-node/commit/f868f53ceb0736a7ba3be7247e990ad0a13195b5
But flow error comes from promtheus.js while other version(0.10, 4) is being tested.
```
 14: import { Counter as PromCounter, Gauge as PromGauge } from 'prom-client';
                                                                ^^^^^^^^^^^^^ prom-client. Required module not found
```
[Travis build result](https://travis-ci.org/eundoosong/jaeger-client-node/builds/392337016?utm_source=github_status&utm_medium=notification)
we can ignore this by using `// $FlowIgnors`, it causes another error in 6+.
Do you have any idea how to avoid?

Reply:
It is feasible, let me try.